src/live_restore/live_restore_server.c

```diff
@@ -441,6 +457,6 @@ __wt_live_restore_server_destroy(WT_SESSION_IMPL *session)
         __live_restore_work_queue_drain(session);
         __wt_spin_destroy(session, &server->queue_lock);
     }
-    __wt_free(session, server);
+    __wt_free(session, S2C(session)->live_restore_server);
```

Comment:
Surely we could just set `S2C(session)->live_restore_server` to NULL on the next line rather that this somewhat odd usage? Couldn't the compiler also optimize this in a weird way?

Reply:
I started with that approach but I couldn't find any similar uses in the rest of the code base. It's also unusual because because __wt_free is meant to null out the pointer for us. Either way changed in [0e28c9e](https://github.com/wiredtiger/wiredtiger/pull/11797/commits/0e28c9e163a4736f9c36ffd7512ee0cc357810df)